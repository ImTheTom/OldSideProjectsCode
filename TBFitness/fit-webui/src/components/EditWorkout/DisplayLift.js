import React, { Component } from "react";
import { connect } from "react-redux";
import { workout_edit_lift } from "../../actions/fit";
import { Button } from "antd";

class DisplayLift extends Component {
	constructor() {
		super();
		this.onEditLift = this.onEditLift.bind(this);
	}

	onEditLift = () => {
		this.setState(this.props.workoutEditLift(this.props.workoutIndex, this.props.liftIndex, this.props.type));
	}

	render() {
		return (
			<div>
				<div className="edit-option">
					{this.props.lifts.name}
					<Button className="display-lift-button" type="primary"  onClick={() => this.onEditLift()}>{this.props.type}</Button>
				</div>
			</div>
		);
	}
}

const mapDispatchToProps = (dispatch) => ({
	workoutEditLift: (workoutIndex, liftIndex, type) => dispatch(workout_edit_lift(workoutIndex, liftIndex, type)),
});

export default connect(undefined, mapDispatchToProps)(DisplayLift);