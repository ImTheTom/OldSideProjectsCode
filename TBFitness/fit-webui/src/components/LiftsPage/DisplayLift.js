import React, { Component } from "react";
import { connect } from "react-redux";
import { edit_lift_name, edit_lift, save_lift, delete_lift } from "../../actions/fit";
import { Input, Button } from "antd";

class DisplayLift extends Component {
	constructor() {
		super();
		this.onEditLift = this.onEditLift.bind(this);
		this.handleInputChange = this.handleInputChange.bind(this);
		this.onSaveLift = this.onSaveLift.bind(this);
		this.onDeleteLift = this.onDeleteLift.bind(this);
	}

	onEditLift = (editType, amount) => {
		this.setState(this.props.editLift(editType, this.props.liftIndex, amount));
	}

	handleInputChange = (e) => {
		this.setState(this.props.editLiftName(this.props.liftIndex, e.target.value));
	}

	onSaveLift = () => {
		this.setState(this.props.saveLift(this.props.liftIndex));
	}

	onDeleteLift = () => {
		this.setState(this.props.deleteLift(this.props.liftIndex));
	}

	render() {
		return (
			<div className ="display-lift">
				<div className = "display-lift-name">
					<Input value={this.props.lifts.name}  onChange={e => this.handleInputChange(e)}></Input>
				</div>
				<div className="display-lift-control">
					<Button className="display-lift-button" type="primary" onClick={() => this.onEditLift("DecreaseWeight", 1)}>&lt;</Button>
					{this.props.lifts.weight}
					<Button className="display-lift-button" type="primary" onClick={() => this.onEditLift("IncreaseWeight", 1)}>&gt;</Button>
				</div>

				<div className="display-lift-control">
					<Button className="display-lift-button" type="primary" onClick={() => this.onEditLift("DecreaseReps", 1)}>&lt;</Button>
					{this.props.lifts.reps}
					<Button className="display-lift-button" type="primary" onClick={() => this.onEditLift("IncreaseReps", 1)}>&gt;</Button>
				</div>

				<div className="display-lift-control">
					<Button className="display-lift-button" type="primary" onClick={() => this.onEditLift("DecreaseSets", 1)}>&lt;</Button>
					{this.props.lifts.sets}
					<Button className="display-lift-button" type="primary" onClick={() => this.onEditLift("IncreaseSets", 1)}>&gt;</Button>
				</div>

				<div className="display-lift-control">
					<Button className="display-lift-button" type="primary" onClick={() => this.onSaveLift()}>Save</Button>
					<Button className="display-lift-button" type="primary" onClick={() => this.onDeleteLift()}>Delete</Button>
				</div>
			</div>
		);
	}
}

const mapDispatchToProps = (dispatch) => ({
	editLiftName: (liftIndex, newName) => dispatch(edit_lift_name(liftIndex, newName)),
	editLift: (editType, liftIndex, amount) => dispatch(edit_lift(editType, liftIndex, amount)),
	saveLift: (liftIndex) => dispatch(save_lift(liftIndex)),
	deleteLift: (liftIndex) => dispatch(delete_lift(liftIndex)),
});

export default connect(undefined, mapDispatchToProps)(DisplayLift);