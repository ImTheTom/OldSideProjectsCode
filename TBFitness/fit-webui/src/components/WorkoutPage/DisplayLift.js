import React, { Component } from "react";
import { connect } from "react-redux";
import { edit_workout_lift, save_workout_lift, edit_workout_lift_name, delete_workout_lift } from "../../actions/fit";
import { Button } from "antd";

class DisplayLift extends Component {
	constructor() {
		super();
		this.onEditWorkout = this.onEditWorkout.bind(this);
		this.onSaveWorkout = this.onSaveWorkout.bind(this);
		this.onDeleteWorkout = this.onDeleteWorkout.bind(this);
	}

	onEditWorkout = (editType, amount) => {
		this.setState(this.props.editWorkoutLift(editType, this.props.workoutIndex, this.props.liftIndex, amount));
	}

	onSaveWorkout = () => {
		this.setState(this.props.saveWorkoutLift(this.props.workoutIndex, this.props.liftIndex));
	}

	onDeleteWorkout = () => {
		this.setState(this.props.deleteWorkoutLift(this.props.workoutIndex, this.props.liftIndex));
	}


	render() {
		return (
			<div className ="display-lift">
				<div className = "display-lift-name">{this.props.lifts.name}</div>

				<div className="display-lift-control">
					<Button className="display-lift-button" type="primary" onClick={() => this.onEditWorkout("DecreaseWeight", 1)}>&lt;</Button>
					{this.props.lifts.weight}
					<Button className="display-lift-button" type="primary" onClick={() => this.onEditWorkout("IncreaseWeight", 1)}>&gt;</Button>
				</div>

				<div className="display-lift-control">
					<Button className="display-lift-button" type="primary" onClick={() => this.onEditWorkout("DecreaseReps", 1)}>&lt;</Button>
					{this.props.lifts.reps}
					<Button className="display-lift-button" type="primary" onClick={() => this.onEditWorkout("IncreaseReps", 1)}>&gt;</Button>
				</div>

				<div className="display-lift-control">
					<Button className="display-lift-button" type="primary" onClick={() => this.onEditWorkout("DecreaseSets", 1)}>&lt;</Button>
					{this.props.lifts.sets}
					<Button className="display-lift-button" type="primary" onClick={() => this.onEditWorkout("IncreaseSets", 1)}>&gt;</Button>
				</div>
				<div className="display-lift-control">
					<Button className="display-lift-button" type="primary" onClick={() => this.onSaveWorkout()}>Save</Button>
					<Button className="display-lift-button" type="primary" onClick={() => this.onDeleteWorkout()}>Delete</Button>
				</div>
			</div>
		);
	}
}

const mapDispatchToProps = (dispatch) => ({
	editWorkoutLiftName: (workoutIndex, liftIndex, newName) => dispatch(edit_workout_lift_name(workoutIndex, liftIndex, newName)),
	editWorkoutLift: (editType, workoutIndex, liftIndex, amount) => dispatch(edit_workout_lift(editType, workoutIndex, liftIndex, amount)),
	saveWorkoutLift: (workoutIndex, liftIndex) => dispatch(save_workout_lift(workoutIndex, liftIndex)),
	deleteWorkoutLift: (workoutIndex, liftIndex) => dispatch(delete_workout_lift(workoutIndex, liftIndex))
});

export default connect(undefined, mapDispatchToProps)(DisplayLift);