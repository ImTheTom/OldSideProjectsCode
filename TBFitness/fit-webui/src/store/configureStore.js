import { createStore, combineReducers, applyMiddleware } from "redux";
import thunk from "redux-thunk";
import workout from "../reducers/workout";
import lifts from "../reducers/lifts";
import editWorkout from "../reducers/editWorkout";

const store = createStore(combineReducers({ workout: workout, lifts: lifts, editWorkout: editWorkout }), applyMiddleware(thunk));

export default store;

