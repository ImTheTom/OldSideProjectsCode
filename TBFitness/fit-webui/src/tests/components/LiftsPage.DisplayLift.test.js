import DisplayLift from "../../components/LiftsPage/DisplayLift";
import React from "react";
import { shallow } from "enzyme";
import LiftsBase from "../fixtures/liftsBase";
import store from "../../store/configureStore";
import { Provider } from "react-redux";

test("should render diplay lift component correctly", () => {
	const wrapper = shallow(
		<Provider store={store}>
			<DisplayLift lifts={LiftsBase[0]} liftIndex={0} key={0}/>
		</Provider>
	);
	expect(wrapper).toMatchSnapshot();
});