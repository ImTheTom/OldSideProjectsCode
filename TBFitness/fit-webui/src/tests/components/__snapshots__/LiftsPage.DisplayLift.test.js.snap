// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render diplay lift component correctly 1`] = `
<Connect(DisplayLift)
  liftIndex={0}
/>
`;
